app/controllers/admin/categories_controller.rb

```diff
@@ -25,12 +25,13 @@ def destroy
 
   def new_or_edit
     @categories = Category.find(:all)
-    @category = Category.find(params[:id])
+    @category = Category.new
```

Comment:
It seems like this is redundant, you should only create the new category if there is no params present